synapse/handlers/oidc.py

```diff
@@ -398,6 +478,22 @@ def _validate_metadata(self, m: OpenIDProviderMetadata) -> None:
             # If we're not using userinfo, we need a valid jwks to validate the ID token
             m.validate_jwks_uri()
 
+        if self._config.backchannel_logout_enabled:
+            if not m.get("backchannel_logout_supported", False):
+                logger.warning(
+                    "OIDC Back-Channel Logout is enabled for issuer %r"
+                    "but it does not advertise support for it",
+                    self.issuer,
+                )
+
+            if not m.get("backchannel_logout_session_supported", False):
```

Comment:
I think this needs to be an `elif` since the warning below says that back-channel logout is enabled.